src/main/java/webApp/webController.java

```diff
@@ -25,43 +30,120 @@ public String index(Model model){
     @PostMapping(value = "/createSurvey", consumes = "application/json", produces = "application/json")
     @ResponseBody
     public Response createSurvey(@RequestBody SurveyMessage surveyMessage) {
-        Survey survey = new Survey(surveyMessage.getName());
+
         Collection<Question> questionList = new ArrayList<>();
 
-        for (QuestionMessage question : surveyMessage.getQuestions()) {
-            if (question.getType().equals("openEnded")) {
-                questionList.add(new OpenEndedQuestion(question.getQuestion()));
-            } else if(question.getType().equals("numberQuestion")) {
-                questionList.add(new NumberQuestion(question.getQuestion(), question.getMin(), question.getMax()));
+        ArrayList<String> questionTexts = new ArrayList<>();
+
+        if (surveyMessage.getQuestions() == null) {
+            return new Response(null, "error", "Questions were null");
+        }
+
+        for (QuestionMessage questionMsg : surveyMessage.getQuestions()) {
+            if (questionMsg == null) {
+                return new Response(null, "error", "Question was null");
             }
+            questionTexts.add(questionMsg.getQuestion());
+            if (questionMsg.getType().equals(openQuestionType)) {
```

Comment:
A switch-case statement may be better suited for this situation. Currently, only 2 types so no issue, but when we have like 5+ types of questions a switch case statement is more efficient and faster.